### DISCUSSION TITLE
Wasm bindgen integration

### DISCUSSION THREAD

**Discussion by MinaMatta98** [leptos-rs/leptos#2183] [Q&A] [answered]
Hello,

Was there any example of wasm bindgen integration with external js libraries and loading the resulting script into a view?

**Comment by gbj** [comment]
There's not really anything magical: Using [wasm-bindgen docs](https://rustwasm.github.io/docs/wasm-bindgen/examples/import-js.html) to import from JS, and then calling it from Rust. Is there a specific example I can help with?

**Reply by MinaMatta98** [reply]
Thank you for coming back to me :).

Take this fn:

```rust
#[wasm_bindgen]
pub fn init_autocomplete(
    map_element: &web_sys::HtmlElement,
    start_search: &web_sys::HtmlElement,
    end_search: &web_sys::HtmlElement,
) {
// Implementation
}
```

Let's say that a view is rendered with the following string:

```html
<script async src=format!("https://maps.googleapis.com/maps/api/js?key={}&libraries=places&callback={callback}&v=beta", resource) defer>
```

Is there a way to call ``` init_autocomplete() ``` as a callback (as demonstrated within the script html query)?

**Reply by gbj** [reply, marked as answer]
Do you mean, is there a way to detect when that other script has loaded, and then call `init_autocomplete` when it has? I believe `<script>` tags have a `load` event, so you can do `on:load`.

**Reply by MinaMatta98** [reply]
Ah, yes...

Thank you
